### PR TITLE
DEV-3884-Add Solution & Test: Rect Perimeter exam function

### PR DESCRIPTION
**Pull request by mohamedLazyBob**
resolves 01-edu/public#1149

### CODE REVIEW

**Review by Hamzaelkhatri — APPROVED** [review]
all is good 💯

**Review by zainabdnaya — COMMENTED** [review]
Well Done

**Review by zainabdnaya — COMMENTED** [review]
All Good

File: tests/rectperimeter_test/main.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+)
+
+func main() {
+	args := [][]int{
+		{1, 2},
+		{0, 0},
+		{8423, 9485},
+		{-1, -1},
+		{2147483647, 3},
+		{14342, -1},
```
Comment (davidrobert99) [inline tests/rectperimeter_test/main.go:17]: Consider adding a test case with a negative value on the left and a positive value on the right. If the student only forgets to verify if w < 0 the test will pass and it should not.

**Review by davidrobert99 — CHANGES_REQUESTED** [review]

**Review by jotapero — APPROVED** [review]
Looks Good :alien:

**Review by tiagocollot — CHANGES_REQUESTED** [review]

File: tests/rectperimeter_test/main.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	}
+	for _, arg := range args {
+
```
[the remark this answers is not in the document]
Comment (jotapero) [reply tests/rectperimeter_test/main.go]: fixed

**Review by tiagocollot — APPROVED** [review]
LGTM ⚡
